SharpPcap/LibPcap/PcapDevice.cs

```diff
@@ -424,7 +440,10 @@ protected virtual RawCapture MarshalRawPacket(IntPtr /* pcap_pkthdr* */ header,
             // marshal the header
             var pcapHeader = PcapHeader.FromPointer(header);
 
-            var pkt_data = new byte[pcapHeader.CaptureLength];
+            byte[] pkt_data = this.m_data_allocator != null
+                ? this.m_data_allocator.Invoke(pcapHeader.CaptureLength)
+                : new byte[pcapHeader.CaptureLength];
+
             Marshal.Copy(data, pkt_data, 0, (int)pcapHeader.CaptureLength);
 
             p = new RawCapture(LinkType,
```

Comment:
@chmorgan Is there a way to change the RawCapture implimentation, to support using arrays larger than `pcapHeader.CaptureLength` from a shared pool.
Since here using the default Shared ArrayPool will result in an extra padding

> The shared pool provides a default implementation of the ArrayPool<T> class that's intended for general applicability. A shared class maintains arrays of multiple sizes, and may hand back a larger array than was actually requested, but it will never hand back a smaller array than was requested. Renting a buffer from a shared class using the Rent method will result in an existing buffer being taken from the pool if an appropriate buffer is available or in a new buffer being allocated if one is not available.

See: https://docs.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.shared?view=netcore-2.2

Reply:
Maybe use the `ByteArraySegment`